module/zfs/dmu_zfetch.c

```diff
@@ -303,10 +289,10 @@ dmu_zfetch(zfetch_t *zf, uint64_t blkid, uint64_t nblks, boolean_t fetch_data,
 	 * A fast path for small files for which no prefetch will
 	 * happen.
 	 */
-	if (zf->zf_dnode->dn_maxblkid < 2) {
+	if ((maxblkid = zf->zf_dnode->dn_maxblkid) < 2) {
```

Comment:
[nit] I think it's easier to read side-effect-free `if` statements
```suggestion
	uint64_t maxblkid = zf->zf_dnode->dn_maxblkid;
	if (maxblkid  < 2) {
```

Reply:
Whatever. Done.